src/main/resources/avro/references.avdl

```diff
@@ -27,8 +105,22 @@ record Reference {
   string name;
 
   /**
+  Start and end locations of how this reference fits into the existing graph.
+  May be null only for a primary sequence that is not joined onto any other.
+  Note that the referenceId for start and end may not be the same.
+  */
+  union { null, ReferenceJoinLocation } startJoin, endJoin;
+
+  /**
+  The sequence to insert between startJoin and endJoin. Must be set if sourceURI
+  is null.
+  */
+  union { null, string } sequence = null;
```

Comment:
I agree that the URL-to-FASTA approach for transferring long sequences to the client leaves a lot to be desired. However, I don't think it's a problem unique to the graph implementation.

I think we should work on this problem against the main branch, and use the string-or-FASTA setup on the graph branch (selecting strings for short sequences and FASTA URLs for those that are too long) until `getSequence` is ready. I've made issue #230 for this.